tpm/tpm.go

```diff
@@ -32,13 +34,27 @@ type Closer interface {
 // extensive resource management that the kernel already handles for us
 // when using the kernel resource manager.
 func Open(path string) (Closer, error) {
-	switch path {
-	case "/dev/tpmrm0":
+	switch {
+	case isDevNode(path, "tpmrm"):
 		return linuxtpm.Open(path)
-	case "/dev/tpm0":
+	case isDevNode(path, "tpm"):
 		slog.Warn("direct use of the TPM can lead to resource exhaustion, use a TPM resource manager instead")
 		return linuxtpm.Open(path)
 	default:
 		return nil, fmt.Errorf("unsupported TPM device path: %s", path)
 	}
 }
+
+func isDevNode(path, kind string) bool {
```

Comment:
thinking back about this, go-fdo-client has same hardcoded logic for TPM devices, so this could open an exposed function that we also use over there w/o reimplementing it